readme.md

```diff
@@ -249,6 +251,38 @@ import adapter from '@flopflip/launchdarkly-adapter';
 </ConfigureFlopFlip>;
 ```
 
+Whenever your application "gains" certain information (e.g. with `react-router`) only further
+down the tree but that information should be used for user targeting (through `adapterArgs.user`) you
+can use `ReconfigureFlopflip`. `ReconfigureFlopflip` itself communicates with `ConfigureFlopflip`
+to reconfigure the given adapter for more fine grained targeting with the pased `user`.
+You also do not have to worry about rendering any number of `ReconfigureFlopflip`s before the adapter is
+initialized (e.g. LaunchDarkly). Requested reconfigurations will be queued and processed once the adapter is ready.
+
+Imagine having `ConfigureFlopflip` above a given component wrapped by a `Route`:
+
+```jsx
+<Route
+  exact={false}
+  path="/:projectKey"
+  render={routerProps => (
+    <React.Fragment>
+      <MyRouteComponent />
+      <ReconfigureFlopflip
+        exact={false}
+        // Note: this should be memoised to not trigger wasteful `reconfiguration`s.
```

Comment:
Good point! If this is important, you should mention this more clearly in the documentation (non only in the example). Is it possible maybe to do this in the internal implementation, so the user does not have to worry about it?

Reply:
Hm, not sure about the implementation but. I guess one would have to deep equal the user prop. Not sure I want that as I rather would like to "encourage" best practices somewhat.